### ISSUE TITLE
"Anonymous function can be static" false positive

### ISSUE THREAD

**Issue by reedy** [hammer-tools/php-hammer#25]
https://gerrit.wikimedia.org/r/c/mediawiki/core/+/951557/1/includes/jobqueue/JobQueueDB.php results in a complaint from Phan, due to errors that would occur in PHP 8.0 (but "ok" in PHP 7.4)

```
19:49:08 includes/jobqueue/JobQueueDB.php:548 PhanStaticCallToNonStatic Static call to non-static method \JobQueue::doDeduplicateRootJob() defined at includes/jobqueue/JobQueue.php:502. This is an Error in PHP 8.0+.
```

**Comment by fMads** [comment]
I have the same issue - I'm pretty sure the issue is, that the closure calls parent inside.
And that's not allowed if the closure is static

**Comment by rentalhost** [comment]
@fMads Hey, could you please provide a more detailed explanation? It would be really helpful if you could include a code example because I'm having some trouble grasping the issue here.

**Comment by reedy** [comment]
There's code in the gerrit link... MediaWiki code base is open source and available for anyone... :)

**Comment by rentalhost** [comment]
Thank you for your time in reporting this issue. The fix has been implemented in **version 0.25.0** and is **available exclusively** through the [**Releases**](https://github.com/hammer-tools/php-hammer/releases) page. It will be implemented in the JetBrains Marketplace once [**this issue**](https://github.com/hammer-tools/php-hammer/issues/10) that are affecting some users are resolved (no ETA).

**In response to your report:**

- The issue has been identified and fixed. Now, when a function's body statically calls non-static methods (including cases like `parent::call()`), it will no longer suggest using `static` for the function since this feature was removed in PHP 8 and later. In the future, an inspection will be implemented to check for issues like this in reverse (#34).
